unittest/test_InodeCache.cpp

```diff
@@ -0,0 +1,192 @@
+// Copyright (C) 2020 Joel Rosdahl and other contributors
+//
+// See doc/AUTHORS.adoc for a complete list of contributors.
+//
+// This program is free software; you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the Free
+// Software Foundation; either version 3 of the License, or (at your option)
+// any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along with
+// this program; if not, write to the Free Software Foundation, Inc., 51
+// Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+#include "../src/Config.hpp"
+#include "../src/Context.hpp"
+#include "../src/InodeCache.hpp"
+#include "../src/Util.hpp"
+#include "../src/hash.hpp"
+
+#include "third_party/catch.hpp"
+
+#ifdef INODE_CACHE_SUPPORTED
+
+namespace {
+
+struct digest
+digest_from_string(const char* s)
+{
+  struct digest digest;
+  struct hash* hash = hash_init();
+  hash_string(hash, s);
+  hash_result_as_bytes(hash, &digest);
+  hash_free(hash);
+  return digest;
+}
+
+bool
+digest_equals_string(const struct digest& digest, const char* s)
+{
+  struct digest rhs = digest_from_string(s);
+  return digests_equal(&digest, &rhs);
+}
+
+bool
+put(const Context& ctx, const char* filename, const char* s, int return_value)
+{
+  return ctx.inode_cache.put(filename,
+                             InodeCache::ContentType::code,
+                             digest_from_string(s),
+                             return_value);
+}
+
+} // namespace
+
+TEST_CASE("Test disabled")
+{
+  Context ctx;
+  ctx.config.set_debug(true);
+  ctx.config.set_inode_cache(false);
+
+  struct digest digest;
+  int return_value;
+
+  CHECK(!ctx.inode_cache.get(
+    "a", InodeCache::ContentType::code, &digest, &return_value));
+  CHECK(!ctx.inode_cache.put(
+    "a", InodeCache::ContentType::code, digest, return_value));
+  CHECK(ctx.inode_cache.get_hits() == -1);
+  CHECK(ctx.inode_cache.get_misses() == -1);
+  CHECK(ctx.inode_cache.get_errors() == -1);
+}
+
+TEST_CASE("Test lookup nonexistent")
+{
```

Comment:
Since the PR was started I have added a `TestUtil::TestContext` class which would be good to use in these test cases as well. A `TestUtil::TestContext` object makes sure to run each test case in a fresh subdirectory to avoid the tests messing with each others' files.

Reply:
Added TestUtil::TestContext to the test cases.